engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java

```diff
@@ -158,26 +157,45 @@ private List<NotificationHistory> sendEmail(Event event, Set<Endpoint> endpoints
         processedEmailCount.increment();
         Action action = event.getAction();
 
-        TemplateInstance subject;
-        TemplateInstance body;
+        final TemplateInstance subject;
+        final TemplateInstance body;
 
-        if (useTemplatesFromDb) {
+        if (featureFlipper.isUseTemplatesFromDb()) {
             Optional<InstantEmailTemplate> instantEmailTemplate = templateRepository
                     .findInstantEmailTemplate(event.getEventType().getId());
-            if (instantEmailTemplate.isEmpty()) {
-                return Collections.emptyList();
-            } else {
+            if (instantEmailTemplate.isPresent()) {
```

Comment:
The old "if template is empty then return Collection.emptyList()" could be restored here with your latest changes to the repository.